catalog_tools/utils/binning.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import decimal
+
+
+def normal_round_to_int(x: float) -> int:
+    """
+    Rounds a float number x to the closest integer.
+
+    Args:
+        x: decimal number that needs to be rounded
+
+    Returns:
+        Rounded value of the given number.
+    """
+
+    sign = 1 if (x >= 0) else -1
+    x = abs(x)
+
+    if x - np.floor(x) < 0.5:
+        return int(sign * np.floor(x))
+    return int(sign * np.ceil(x))
+
+
+def normal_round(x: float, n: int) -> float:
+    """
+    Rounds a float number x to n number of decimals. If the number
+    of decimals is not given, we round to an integer.
+
+    Args:
+        x: decimal number that needs to be rounded
+        n: number of decimals, optional
+
+    Returns:
+        Value rounded to the given number of decimals.
+    """
+
+    power = 10**n
+    return normal_round_to_int(x * power) / power
+
+
+def bin_to_precision(x: float, delta_m: float = 0.1) -> float:
```

Comment:
would call it delta_x

Reply:
I will, delta_m is what I'm used to since it represents magnitudes and is called so in papers etc., but in this context for the sake of consistency you're right. In the future, maybe it might be better to change x to m?